manifests/install.pp

```diff
@@ -1,4 +1,11 @@
-# Private define
+# @summary Private define
+#
+# @param gem_name
+# @param provider
+# @param gem_version
+# @param gem_source
+# @param gem_install_options
+#
```

Comment:
No need to document parameters of internal API.  We have a `@api` parameter for this:

```suggestion
# @api private
```

It may make sense to add a `assert_private()` at the beginning of the define to make sure we are not being called by the end-user directly.

Reply:
will do, didnt know that was a possibility :)